perceval/runtime/remote_job.py

```diff
@@ -199,4 +194,25 @@ def get_results(self) -> Any:
                     res["results"] = result_mapping_function(res["results"], **self._delta_parameters)
             else:
                 results["results"] = result_mapping_function(results["results"], **self._delta_parameters)
+
+    def get_results(self) -> Dict:
+        job_status = self.status
+        if not job_status.completed and not job_status.maybe_completed:
+            raise RuntimeError('The job is still running, results are not available yet.')
+
+        if not self.is_maybe_complete and job_status.status != RunningStatus.SUCCESS:
+            raise RuntimeError(f'The job failed: {job_status.stop_message}')
+
+        results = None
+        if job_status.status == RunningStatus.SUCCESS:
+            response = self._rpc_handler.get_job_results(self._id)
+            results = deserialize(json.loads(response)['results'])
+        elif self.is_maybe_complete:
+            warnings.warn("Partial or unknown job status, trying to get result anyway.")
+            try:
+                response = self._rpc_handler.get_job_results(self._id)
+                results = deserialize(json.loads(response)['results'])
+            except KeyError:
+                return {}
```

Comment:
Same as for the local job: if we return an empty dictionary, we should as well raise an exception.